python-template/{{cookiecutter.placeholder_repo_name}}/setup.py

```diff
@@ -36,17 +36,43 @@ def load_requirements(*requirements_paths):
     with -c in the requirements files.
     Returns a list of requirement strings.
     """
+    by_canonical_name = {}
+
+    def check_name_consistent(package):
+        """
+        Raise exception if package is spelled different ways.
+
+        This ensures that packages are spelled consistently so we can match
+        constraints to packages. It also ensures that if we require a package
+        with extras we don't constrain it without mentioning the extras (since
+        that too would interfere with matching constraints.)
+        """
+        canonical = package.lower().replace('_', '-').split('[')[0]
+        seen_spelling = by_canonical_name.get(canonical)
+        if seen_spelling is None:
+            by_canonical_name[canonical] = package
+        elif seen_spelling != package:
+            raise Exception(
+                f'Encountered both "{seen_spelling}" and "{package}" in requirements '
+                'and constraints files; please use just one or the other.'
+            )
+
     requirements = {}
     constraint_files = set()
 
     # groups "pkg<=x.y.z,..." into ("pkg", "<=x.y.z,...")
```

Comment:
can we update the comment to include how the regex treats [] ? it's a pretty hard regex to read